pyvista/core/utilities/geometric_objects.py

```diff
@@ -383,6 +396,524 @@ def Sphere(
+        points. Must have at least two points.  Can have any value as long
+        as range is within 360 degrees. Large magnitudes may result in
+        problems with endpoint overlap detection.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying ``phi``
+        points. Must have at least two points.  Must be between
+        0 and 180 degrees.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center coordinate vector in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction coordinate vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees ``phi``.
+
+    radians : bool, default: False
+        Whether to use radians for ``theta`` and ``phi``. Default is degrees.
+
+    tol_radius : float, default: 1.0e-8
+        Absolute tolerance for endpoint detection for ``radius``.
+
+    tol_angle : float, optional
+        Absolute tolerance for endpoint detection
+        for ``phi`` and ``theta``. Unit is determined by choice
+        of ``radians`` parameter.  Default is 1.0e-8 degrees or
+        1.0e-8 degrees converted to radians.
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Solid sphere mesh.
+
+    See Also
+    --------
+    pyvista.SolidSphere: Sphere creation using linear sampling.
+    pyvista.Sphere: Sphere that describes outer 2D surface.
+
+    Examples
+    --------
+    Linearly sampling spherical coordinates does not lead to
+    cells of all the same size at each radial position.
+    Cells near the poles have smaller sizes.
+
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> solid_sphere = pv.SolidSphereGeneric(
+    ...     radius=np.linspace(0, 0.5, 2),
+    ...     theta=np.linspace(180, 360, 30),
+    ...     phi=np.linspace(0, 180, 30),
+    ... )
+    >>> solid_sphere = solid_sphere.compute_cell_sizes()
+    >>> solid_sphere.plot(
+    ...     scalars="Volume", show_edges=True, clim=[3e-5, 5e-4]
+    ... )
+
+    Sampling the polar angle in a nonlinear manner allows for consistent cell volumes.  See
+    `Sphere Point Picking <https://mathworld.wolfram.com/SpherePointPicking.html>`_.
+
+    >>> phi = np.rad2deg(np.arccos(np.linspace(1, -1, 30)))
+    >>> solid_sphere = pv.SolidSphereGeneric(
+    ...     radius=np.linspace(0, 0.5, 2),
+    ...     theta=np.linspace(180, 360, 30),
+    ...     phi=phi,
+    ... )
+    >>> solid_sphere = solid_sphere.compute_cell_sizes()
+    >>> solid_sphere.plot(
+    ...     scalars="Volume", show_edges=True, clim=[3e-5, 5e-4]
+    ... )
+
+    """
+    if radius is None:
+        radius = np.linspace(0, 0.5, 5)
+    radius = np.asanyarray(radius)
+
+    # Default tolerance from user is set in degrees
+    # But code is in radians.
+    if tol_angle is None:
+        tol_angle = np.deg2rad(1e-8)
+    elif not radians:
+        tol_angle = np.deg2rad(tol_angle)
```

Comment:
Codecov claims this line is untested. I'm chronically on mobile so I can't judge if this is true or whether it's an issue.

Reply:
Should be tested with https://github.com/pyvista/pyvista/pull/4965/commits/ce93559d1db7dba891d0f2fb729d094b10e63d7e